src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ScannerZipInstaller.java

```diff
@@ -13,33 +13,38 @@
 import java.nio.charset.Charset;
 import java.security.cert.Certificate;
 
+import com.synopsys.integration.blackduck.api.core.BlackDuckResponse;
+import com.synopsys.integration.blackduck.api.core.response.UrlSingleResponse;
+import com.synopsys.integration.blackduck.http.BlackDuckRequestBuilder;
+import com.synopsys.integration.blackduck.http.client.BlackDuckHttpClient;
+import com.synopsys.integration.blackduck.service.request.BlackDuckRequest;
 import org.apache.commons.compress.archivers.ArchiveException;
 import org.apache.commons.io.FileUtils;
-import org.apache.commons.lang3.StringUtils;
 
 import com.synopsys.integration.blackduck.exception.BlackDuckIntegrationException;
 import com.synopsys.integration.blackduck.http.client.SignatureScannerClient;
 import com.synopsys.integration.blackduck.keystore.KeyStoreHelper;
-import com.synopsys.integration.blackduck.service.dataservice.BlackDuckRegistrationService;
 import com.synopsys.integration.exception.IntegrationException;
 import com.synopsys.integration.log.IntLogger;
 import com.synopsys.integration.rest.HttpUrl;
-import com.synopsys.integration.rest.request.Request;
 import com.synopsys.integration.rest.response.Response;
 import com.synopsys.integration.util.CleanupZipExpander;
 import com.synopsys.integration.util.OperatingSystemType;
 
 public class ScannerZipInstaller implements ScannerInstaller {
-    public static final String DEFAULT_SIGNATURE_SCANNER_DOWNLOAD_URL_SUFFIX = "download/scan.cli.zip";
-    public static final String WINDOWS_SIGNATURE_SCANNER_DOWNLOAD_URL_SUFFIX = "download/scan.cli-windows.zip";
-    public static final String MAC_SIGNATURE_SCANNER_DOWNLOAD_URL_SUFFIX = "download/scan.cli-macosx.zip";
+    private static final String PLATFORM_PARAMETER_KEY = "platforms";
+    private static final String MAC_PLATFORM_PARAMETER_VALUE = "macosx";
+    private static final String LINUX_PLATFORM_PARAMETER_VALUE = "linux";
+    private static final String WINDOWS_PLATFORM_PARAMETER_VALUE = "windows";
+
+    private static final String SCAN_CLI_TOOL_DOWNLOAD_URL = "api/tools/scan.cli.zip/versions/latest/";
 
     public static final String BLACK_DUCK_SIGNATURE_SCANNER_INSTALL_DIRECTORY = "Black_Duck_Scan_Installation";
-    public static final String VERSION_FILENAME = "blackDuckVersion.txt";
+    public static final String SCAN_CLI_VERSION_FILENAME = "scan-cli-version.txt";
 
     private final IntLogger logger;
-    private final SignatureScannerClient signatureScannerClient;
-    private final BlackDuckRegistrationService blackDuckRegistrationService;
+    private final SignatureScannerClient signatureScannerClient; // TOME maybe nuke?
```

Comment:
Testing why the download of the scan.cli.zip fails sometimes (at different places: sometimes during the copying from temporary zip to real directory, and sometimes at the following line that involves the SignatureScannerClient:                 `keyStoreHelper.updateKeyStoreWithServerCertificate(downloadUrl.url().getHost(), serverCertificate, scanPaths.getPathToCacerts());`)